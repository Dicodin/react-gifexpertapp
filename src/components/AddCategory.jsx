// Iniciando todo con el snnipet "rafc"
import React, { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

    // El estado se inicia con "inputValue = ''"
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // Para igualar nuestro estado a lo que escribamos en el input
        setInputValue(e.target.value);

    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // De esta forma si podemos mutar el estado del arreglo que viene "dentro" de la función de nuestro prop "setCategories"
            setCategories(categ => [inputValue, ...categ]);
            // Esta es la función que llamamos de nuestro estado para "limpiar" nuestro input
            setInputValue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder='Buscar gifs'
                value={inputValue}
                onChange={handleInputChange}
            ></input>
        </form>
    )
}

// Validamos que nuestro prop "setCategories" sea del tipo correspondiente "func" y que sea requerido
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
