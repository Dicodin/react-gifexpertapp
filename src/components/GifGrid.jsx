import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    // Uso mi "custom hook" "useFecthGifs"
    const { data: images, loading } = useFecthGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {/* Para abreviar la condici√≥n "if" con un solo resultado */}
            {loading && <p className="animate__animated animate__flash">Loading</p>}

            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id} // Cada elemento debe tener su key
                            {...img} // Usamos el operador spread para enviar todas las props
                        ></GifGridItem>
                    ))
                }
            </div>
        </>
    )
}