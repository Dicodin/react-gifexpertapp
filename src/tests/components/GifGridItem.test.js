import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas sobre <GifGridItem></GifGridItem>', () => {

            const title = 'titulo';
            const url = 'https://localhost/imagen.jpg';
            const wrapper = shallow( < GifGridItem title = { title }
                url = { url } > < /GifGridItem>);

                test('debe de mostrar de forma correcta el componente ', () => {

                    expect(wrapper).toMatchSnapshot();

                });

                test('debe de tener un title ', () => {

                    const p = wrapper.find('p');
                    expect(p.text().trim()).toBe(title);

                });

                test('debe de tener la imagen igual al url y alt de los props ', () => {

                    const img = wrapper.find('img');
                    expect(img.prop('src')).toBe(url);
                    expect(img.prop('alt')).toBe(title);

                });

                test('debe de tener la clase animate__fadeIn ', () => {

                    const div = wrapper.find('div');
                    expect(div.hasClass('animate__fadeIn')).toBe(true);

                });


            });