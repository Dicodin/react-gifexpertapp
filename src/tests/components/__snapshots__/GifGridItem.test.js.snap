// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGridItem></GifGridItem> debe de mostrar de forma correcta el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://localhost/imagen.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
